packages/abi-gen/src/index.ts

```diff
@@ -116,12 +126,111 @@ function registerPythonHelpers(): void {
             return new Handlebars.SafeString(
                 `\n${cliFormat.wrap(devdocDetails || '', {
                     paddingLeft: ' '.repeat(indent),
-                    width: columnsPerRow - indent,
+                    width: columnsPerRow,
                     ansi: false,
                 })}\n`,
             );
         },
     );
+    Handlebars.registerHelper(
+        'makeParameterDocstringRole',
+        (name: string, description: string, indent: number, options) => {
+            let docstring = `:param ${name}:`;
+            if (description && description.length > 0) {
+                docstring = `${docstring} ${description}`;
+            }
+            return new Handlebars.SafeString(utils.wrapPythonDocstringRole(docstring, indent));
+        },
+    );
+    Handlebars.registerHelper(
+        'makeReturnDocstringRole',
+        (description: string, indent: number, options) =>
+            new Handlebars.SafeString(utils.wrapPythonDocstringRole(`:returns: ${description}`, indent)),
+    );
+    Handlebars.registerHelper(
+        'makeEventParameterDocstringRole',
+        (eventName: string, indent: number, options) =>
+            new Handlebars.SafeString(
+                utils.wrapPythonDocstringRole(
+                    `:param tx_hash: hash of transaction emitting ${eventName} event`,
+                    indent,
+                ),
+            ),
+    );
+    Handlebars.registerHelper('tupleDefinitions', (abisJSON: string, options) => {
+        const abis: AbiDefinition[] = JSON.parse(abisJSON);
+        // build an array of objects, each of which has one key, the Python
+        // name of a tuple, with a string value holding the Python
+        // definition of that tuple.  using a key-value object conveniently
+        // filters duplicate references to the same tuple.
+        const tupleDefinitions: { [pythonTupleName: string]: string } = {};
```

Comment:
Nit: Let's use type `ObjectMap<string>` here. It's in `@0x/types`

Reply:
Honestly I prefer this way because

1. you can see everything right here (don't have to go look up what an `ObjectMap` is),
2. `ObjectMap` itself is a trivial enough to just inline here without even adding another line, and
3. I think having the index signature name visible here, `pythonTupleName`, really helps improve the readability of this somewhat convoluted code.